src/node_buffer.cc

```diff
@@ -35,6 +55,12 @@
   CHECK_NOT_OOB(end <= end_max);                                            \
   size_t length = end - start;
 
+// used by the ReadInt64 functions to determine whether to return a Number
+// or String, based on whether or not a JS Number will lose precision.
+// http://stackoverflow.com/q/307179/376773
+#define JS_MAX_INT +9007199254740992LL
+#define JS_MIN_INT -9007199254740992LL
```

Comment:
Both of these should be `9007199254740991` instead, since `var n = 9007199254740992; console.log(n + 1);` will output `9007199254740992` also.